code/lib/core-common/src/utils/load-main-config.ts

```diff
@@ -12,10 +12,10 @@ export async function loadMainConfig({
 }): Promise<StorybookConfig> {
   await validateConfigurationFiles(configDir);
 
-  const mainJsPath = path.resolve(configDir, 'main');
+  const mainJsPath = serverResolve(path.resolve(configDir, 'main')) as string;
```

Comment:
Why override the type here?  It's possible that the `main` is not found, right?  In which case `null` is returned, there's even a truthy check for it below.

Reply:
if it's not found, an error will be thrown in `validateConfigurationFiles`

Reply:
Maybe worth a code comment to that effect, then?

Reply:
Sounds great, I'll do it in a second iteration!